crates/sui-framework/packages/sui-framework/sources/random.move

```diff
@@ -129,4 +131,146 @@ module sui::random {
     ) {
         update_randomness_state(self, new_round, new_bytes, ctx);
     }
+
+
+    /// Unique randomness generator, derived from the global randomness.
+    struct RandomGenerator has drop {
+        seed: vector<u8>,
```

Comment:
Should we use `u256` for the seed? I see that it's the output of SHA256, so this would be a more type-safe/concise representation (a `vector` carries an extra length byte).

More generally, maybe we want a version of `hmac_sha3_256` that returns a `u256`?

Reply:
HMAC by definition should accept any vector of bytes as the key, so I think that even if we change its output to u256 and use it as the seed here, we'd still need to convert that u256 seed to s vector in `derive_next_block` below, overall costing more gas.